src/components/SearchBar.tsx

```diff
@@ -0,0 +1,52 @@
+import Image from 'next/image';
+import { useRouter } from 'next/navigation';
+import { useSearchParams } from 'next/navigation';
+import { useState } from 'react';
+
+function SearchBar() {
+	const [isOpen, setIsOpen] = useState(false);
+	const router = useRouter();
+	const [query, setQuery] = useState('');
+
+	const handleKeyDown = (event: React.KeyboardEvent<HTMLInputElement>) => {
+		if (event.key === 'Enter') {
+			router.push(`?query=${encodeURIComponent(query)}`);
+		}
+	};
+
+	return (
+		<>
+			<div className={`self-stretch h-auto px-5 py-4  bg-neutral-900 flex-col justify-center items-center gap-2.5 flex  transition-all duration-300 ease-in-out`}>
+				<div
+					onClick={() => {
+						setIsOpen(!isOpen);
```

Comment:
If we mutate state in the "setState" function we should work on the "previous" value like `
setIsOpen( prev => !prev );
`it's a common mistake in React `
setIsOpen( prev => !prev );
` it's a common mistake in React

Reply:
good note, will correct it